### PR TITLE
Add Woocomerce Onboarding task to set up Stripe

### PR DESCRIPTION
**Pull request by JulianBustamante**
### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/dotcom-forge/issues/8257

This PR adds the onboarding task to get paid with Stripe and removes Woo payments.

Screenshot 2024-07-18 at 10 59 07 PM

### How to test the changes in this Pull Request:

1. Create a free trial eCommerce site for Development following the steps described in this https://hostedwoo.wordpress.com/2023/02/09/setting-up-a-free-ecommerce-trial-site-for-development/
2. Open atomic `_cli` and install Stripe plugin `wp wpcomsh plugin use-managed woocommerce-gateway-stripe --activate`
3. Also, from the `_cli` Update the `woocommerce_onboarding_profile` option by using the following command `wp option patch insert woocommerce_onboarding_profile partner stripe`
4. Check out this branch on your local environment and synchronize the files with the plugin on atomic by following these steps. https://teameaglep2.wordpress.com/how-to-guides/working-with-the-woocommerce-calypso-bridge/
5. Go to the `wc-admin` page `/wp-admin/admin.php?page=wc-admin`
6. Confirm that the `Get paid with Stripe` task is displayed.
7. You should be redirected to Stripe Connect.

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by epeicher — APPROVED** [review]
I have tested it, and it works as advertised. After following the test steps, I can see the `Get paid with Stripe` in the onboarding tasks. LGTM!

CleanShot 2024-07-19 at 13 19 14@2x

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

File: includes/free-trial/partners/class-wc-calypso-bridge-partner-stripe.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// Only for free trials.
+		if ( ! wc_calypso_bridge_is_ecommerce_trial_plan() ) {
+			return;
+		}
```
Comment (cpapazoglou) [inline includes/free-trial/partners/class-wc-calypso-bridge-partner-stripe.php:39]: Let's remove this condition, `$onboarding_profile['partner']` should be enough.

Comment (JulianBustamante) [reply includes/free-trial/partners/class-wc-calypso-bridge-partner-stripe.php:39]: I created a [follow-up task](https://github.com/Automattic/dotcom-forge/issues/8418) where we will work on improvements such as removing duplicated code.

**Pull request merged**